### ISSUE TITLE
deleting tracks really quickly causes this error

### ISSUE THREAD

**Issue by CoryWBoris** [CoryWBoris/AutoClip#4]
I don't know why, but deleting tracks very quickly causes this error. But, there is no error if you delete each track and pause for like a second for some reason.

...
INFO:root:866 - Type of clip_slot: <class 'ClipSlot.ClipSlot'>
ERROR:ableton.v2.base.task:929 - Error when executing task
Traceback (most recent call last):  File "output/Live/mac_universal_64_static/Release/python-bundle/MIDI Remote Scripts/ableton/v2/base/task.py", line 235, in do_update  File "output/Live/mac_universal_64_static/Release/python-bundle/MIDI Remote Scripts/a
    if clip_slot.has_clip:Boost.Python.ArgumentError: Python argument types in
    None.None(ClipSlot)
did not match C++ signature:
    None(TPyHandle<AGroupAndClipSlotBase>)

**Comment by CoryWBoris** [comment]
I handled the exceptions as best as I could, as there was really nothing more I could do at this point in time. The script functions exactly as it purports to, the only errors are from rapidly deleting tracks due to a race condition between removing event listeners from the track itself and removing the event listener of a track from the song() properties. I would be really grateful to anyone in the community who could help out with this! you'd be making the script more robust in your contribution. But the exception handling allows for a normal flow and I haven't seen any errors since.

**Comment by CoryWBoris** [comment]
this is the beta version only that you would see an error message, functionality is not interrupted.

**Comment by CoryWBoris** [comment]
Fixed it! Explanation below:

I fixed the error message we were getting by properly dealing with the output of a clip slot in the race condition where the deletion of a track causes some event handlers to not get fully removed at the time we check our clip slots, since changes in clip slots trigger our check of clip slots. Ultimately I fixed the error where the None.None(ClipSlot) is not converted properly to its C++ object, and so now the type properly gets converted to the None pointer in C++ and so we can effectively handle this intermediary step without that blasted red error message of doom. but actually low key that terminal slaps and got me inspired to make remote scripts in the first place. And i am just speculating, as I haven no idea what happens at Ableton. Much love to them though!
